src/components/MovieDetail.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import { BrowserRouter as Router, Route, Link, Redirect} from 'react-router-dom'
+
+
+class MovieDetail extends Component {
+    findMovie = movieId => {
+        return this.props.movies.find(m => m.id == movieId) 
+    }
```

Comment:
When using a component just for visual purposes, you should not make logic like that.
This component should just get some data (-the movie details), and present it, you should not pass all the movies